### ISSUE TITLE
macOS: Sign and notarize the latest release: OpenSCAD-2021.01

### ISSUE THREAD

**Issue by kintel** [openscad/openscad#5421]
The current latest release (OpenSCAD-2021.01) is not signed and notarized.
An attempt to sign and notarize the current binary was done as part of #880, but resulted in a binary which only worked on Intel.

The missing piece is: Why does the signed and notarized 2021.01 binary crash on Apple Silicon, and is there a way of fixing this?

**Comment by ElhemEnohpi** [comment]
I don't know why the signed and notarized version crashes. I do know that if you just sign it, with an ad-hoc signature, you can at least get it to run then, after using the "Open Anyway" button in System Preferences / Security, instead of getting the weird "Finder does not have permission to open (null)" error.  The user can do that on their own, with:

`codesign --deep -f -s - /Applications/OpenSCAD.app`

But that means they have to somehow figure out that they must do that, maybe by coming here. Maybe doing that on the release version, or making a note about it, would at least be an improvement, until the notarization can be figured out?

**Comment by kintel** [comment]
The official 2021.01 binaries have now been replace with ad-hoc signed binaries.

**Comment by ElhemEnohpi** [comment]
Sounds good. But I just tried downloading it - from https://github.com/openscad/openscad/releases/download/openscad-2021.01/OpenSCAD-2021.01.dmg - and it still doesn't have a code signature. The modification date is still January 2021. Tried emptying my browser cache too.

**Comment by kintel** [comment]
oh, we have binaries hosted on github too? I've updated those as well now.

**Comment by ElhemEnohpi** [comment]
Ok, it works now, with "Open Anyway". I also tried to install it with Homebrew, but that now fails with a SHA256 mismatch. I guess the cask code has to be updated for that.

**Comment by kintel** [comment]
Yeah, I guess homebrew would need to update the sha.

**Comment by gingerbeardman** [comment]
I filed a PR: https://github.com/Homebrew/homebrew-cask/pull/193120

**Comment by sciencejoan** [comment]
I tried opening the 2021.01 from the openscad.org/downloads page on a brand new M4 iMac and got the error "Apple could not verify “OpenSCAD-2021.01.app” is free of malware that may harm your Mac or compromise your privacy." The only options were "Done" or "Move to trash." On the bright side, the 2024.11.01 (git 5dd397d7e) opens perfectly fine on the M4 and is blindingly fast.

**Comment by gingerbeardman** [comment]
> I tried opening the 2021.01 from the openscad.org/downloads page on a brand new M4 iMac and got the error "Apple could not verify “OpenSCAD-2021.01.app” is free of malware that may harm your Mac or compromise your privacy." The only options were "Done" or "Move to trash." 

As expected. You still have to right-click > open *twice* and then you'll see:

<img width="372" alt="Screen shot 2024-11-24 at 20 25 27" src="https://github.com/user-attachments/assets/c5485f20-b563-4d5f-9444-55479a7b6e39">

**Comment by sciencejoan** [comment]
I don't get that message unless I open the package contents with a right click, and then go to the folder  Contents >MacOS > OpenSCAD and in turn click on that.   (No amount of right-clicking gives the message you show from the icon on the install disk image, but if I go to the security settings anyway it does give me the option there - without telling me about it. Argh.)  As a side note, we are authors of books that use OpenSCAD to teach math, so my purpose here is to document for our readers what the minimum path is....pretty much we're just telling them to use the 2024 build until this gets more straightforward.

**Comment by gingerbeardman** [comment]
> I filed a PR: [Homebrew/homebrew-cask#193120](https://github.com/Homebrew/homebrew-cask/pull/193120)

this has been approved :)

**Comment by nophead** [comment]
Why on earth does anybody use Macs these days? I certainly never will.

**Comment by curliq** [comment]
This fixed it for me:

Open system settings -> Privacy and Security -> Scroll all the way down -> there's a "Allow OpenSCAD to be opened" button

**Comment by ElhemEnohpi** [comment]
@curliq yes, we know that, thanks. However, it only became possible to do that two weeks ago, when the binaries were updated with a partial fix (ad-hoc signature). Soon there should be a full fix (notarization), so that you won't even have to do that, and just answer "Ok" to the usual "this was downloaded from the Internet. Open it?". That's the point of this issue, to figure out the steps to get notarization to work.

**Comment by copecog** [comment]
> Why on earth does anybody use Macs these days? I certainly never will.
> […](#)

I concur, I don't know why anyone would use anything besides OpenBSD!

**Comment by haty1** [comment]
It is now March 2025 and I just installed OpenSCAD via Homebrew and am still getting this error message when I try and open the app. Seems it is still a problem

**Comment by kintel** [comment]
It's still an issue for the 2021.01 release: Signing the existing binary someone causes it to crash on Intel. We could try to dig out an old Mac and rebuild it from source and see if that helps, but it's a lot of effort for uncertain gain.

This is fixed in development builds since December though, so the issue only affect the 2021.01 release.

**Comment by pkernevez** [comment]
Do you plan to to release a signed and notarized version of the OpenSCAD ?
The idea is not to by pass the OS protection by changing the security settings.
I already signed application for Mx silicon processor and it works fine (and your are able to sign both Intel & Silicon from a silicon mac (and I suppose we should be able to do the inverse).

**Comment by kintel** [comment]
The nightly binaries are signed and notarized. I haven't been able to sign the 2021.01 release without causing it to crash on launch.

**Comment by gingerbeardman** [comment]
The issue is that the "official" build is so old people don't know there are development nightlies.

**Comment by kintel** [comment]
We're working on a new official release, but it will still take a while as our dev team is tiny. Hoping for end of 2025.

If anyone has experience in this area, help would be appreciated.

**Comment by pkernevez** [comment]
> The issue is that the "official" build is so old people don't know there are development nightlies.

It was my case. I use the last nightly build and it works fine.
And to be transparent, I used OpenScad 4 years ago (the last time I needed to do 3D printing) and it worked fine. 
But when I saw that they wasn't any release since 4 years, no perfect support for Mac Silicon and no notarized binary, I start looking for an alternative.
Without this post I didn't found the last build.

You may add a link on the download page to the nightly build.

There isn't a nightly build stable enough to be promoted as a new release ?

I only worked on packaging a java app with notarization. I don't think I could help you.

**Comment by jordanbrown0** [comment]
The downloads page has a section for development snapshots. It's even on the left sidebar.

https://openscad.org/downloads.html#snapshots

**Comment by gingerbeardman** [comment]
I think it would be good to signpost the newer downloads from right next to the current/official download.

"newer versions of OpenSCAD are work in progress but are available < link >"

The phrases "development snapshots" or "nightlies" are known by developers, but mean nothing to the average user.

**Comment by jcheatham** [comment]
I agree with @gingerbeardman , even as a developer I tried the first option, then brew, and was inclined to give up thinking the project was dead... but I refused to give up hope and found this thread.  Development snapshots at the top would have helped me for sure; OR a blurb in the macOS section saying "We know the official version has signing issues, check out the latest snapshots instead" would have helped :)

**Comment by HyperTextGo** [comment]
I installed via homebrew, which was successful*
However when running it with "/opt/homebrew/bin/openscad" in the terminal, it tells me that it cant run.

<img width="251" height="251" alt="Image" src="https://github.com/user-attachments/assets/4b76a216-1eb3-4028-af28-0ceb4e0c2c0b" />
Upon clicking the question mark it pulls up a page on how bypassing this could lead to the computer being infected with malware. 

I don't want to see what happpens when I bypass it

The terminal told me something about roseta 2, the intel interpreter, but as I am on macos 14, I think it was installed

Ill try the nightly builds, but there should be newer release builds because it scare people to download programs that last were updated 4 years ago
